scripts-dev/update_database

```diff
@@ -82,19 +88,21 @@ if __name__ == "__main__":
     # Setup instantiates the store within the homeserver object and updates the
     # DB.
     hs.setup()
-    store = hs.get_datastore()
 
-    async def run_background_updates():
-        await store.db_pool.updates.run_background_updates(sleep=False)
-        # Stop the reactor to exit the script once every background update is run.
-        reactor.stop()
+    if args.run_background_updates:
+        store = hs.get_datastore()
 
-    def run():
-        # Apply all background updates on the database.
-        defer.ensureDeferred(
-            run_as_background_process("background_updates", run_background_updates)
-        )
+        async def run_background_updates():
+            await store.db_pool.updates.run_background_updates(sleep=False)
+            # Stop the reactor to exit the script once every background update is run.
+            reactor.stop()
+
+        def run():
+            # Apply all background updates on the database.
+            defer.ensureDeferred(
+                run_as_background_process("background_updates", run_background_updates)
+            )
 
-    reactor.callWhenRunning(run)
+        reactor.callWhenRunning(run)
 
-    reactor.run()
+        reactor.run()
```

Comment:
maybe better to factor all this stuff out to a separate function, instead of putting it in a big `if` statement?

Reply:
Yep much cleaner - https://github.com/matrix-org/synapse/pull/10954/commits/d3daf7181cb24a556cd16973b984414660ae0b23 (+https://github.com/matrix-org/synapse/pull/10954/commits/6da417509229531e7112530b8bb21b6cd1656ddf).